### PR TITLE
Inherit from Alpine base image. Use OpenJDK

### PR DESCRIPTION
**Pull request by ghost**
@TylerJewell as discussed a lighter Che base image
Signed-off-by: Eugene Ivantsov eivantsov@codenvy.com

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
-EXPOSE 8080
+ADD /assembly/assembly-main/target/eclipse-che-*/eclipse-che-* /home/user/che/
 
-ADD /assembly/assembly-main/target/eclipse-che-*/eclipse-che-* /home/user/che
 ENV CHE_HOME /home/user/che
 
 ENTRYPOINT [ "/home/user/che/bin/che.sh", "-c" ]
-CMD [ "run" ]
+
+CMD [ "run" ]
```
Comment (benoitf) [inline Dockerfile]: missing newline

File: Dockerfile [hunk 2]
```diff
@@ -1,34 +1,40 @@
...
-    sudo apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+FROM alpine:3.4
 
-USER user
+ENV LANG=C.UTF-8 \
+    DOCKER_BUCKET=get.docker.com \
+    DOCKER_VERSION=1.11.2 \
+    DOCKER_SHA256=8c2e0c35e3cda11706f54b2d46c2521a6e9026a7b13c7d4b8ae1f3a706fc55e1 \
+    JAVA_HOME=/usr/lib/jvm/default-jvm/jre
+ENV PATH ${PATH}:${JAVA_HOME}/bin
```
Comment (benoitf) [inline Dockerfile]: could 3 ENV instructions could be merged ?
adding PATH and CHE_HOME /home/user/che

File: Dockerfile [hunk 3]
```diff
@@ -1,34 +1,40 @@
-FROM ubuntu
-RUN apt-get update && apt-get -y install curl sudo procps wget && \
-    echo "%sudo ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
-    useradd -u 1000 -G users,sudo -d /home/user --shell /bin/bash -m user && \
-    echo "secret\nsecret" | passwd user && \
-    curl -sSL https://get.docker.com/ | sh && \
-    # Add user to docker group: 100 and 50 are this gIDs in boot2docker and Docker for Mac
-    usermod -aG docker,100,50 user && \
-    sudo apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+FROM alpine:3.4
```
Comment (benoitf) [inline Dockerfile]: Do we need to add some comments before the FROM to explain how to run this Dockerfile ?

Comment (ghost) [inline Dockerfile]: We have docs explaining how to run

Comment (benoitf) [inline Dockerfile]: maybe add pointer to the doc then ?

Comment (benoitf) [inline Dockerfile]: thx

**Comment by benoitf** [comment]
we may install docker as apk package and not install curl as well

**Comment by benoitf** [comment]
do we still need curl ?

**Pull request closed without merging**